test/integration/authorization_test.go

```diff
@@ -609,55 +612,205 @@ type subjectAccessReviewTest struct {
+				if tmp, err = test.kubeAuthInterface.SubjectAccessReviews().Create(toKubeClusterSAR(test.clusterReview)); err == nil {
+					actualResponse = tmp.Status
+				}
 			}
-			return false, nil
+			testErr := test.kubeErr
+			if len(testErr) == 0 {
+				testErr = test.err
+			}
+			if len(testErr) > 0 {
+				if err == nil {
+					failMessage = fmt.Sprintf("%s: Expected error: %v\ngot\n\t%#v", test.description, testErr, actualResponse)
+					return false, nil
+				} else if !strings.HasPrefix(err.Error(), testErr) {
+					failMessage = fmt.Sprintf("%s: expected\n\t%v\ngot\n\t%v", test.description, testErr, err)
+					return false, nil
+				}
+			} else {
+				if err != nil {
+					failMessage = fmt.Sprintf("%s: unexpected error: %v", test.description, err)
+					return false, nil
+				}
+			}
+
+			if (actualResponse.Allowed != test.response.Allowed) || (!strings.HasPrefix(actualResponse.Reason, test.response.Reason)) {
+				if test.localReview != nil {
+					failMessage = fmt.Sprintf("%s: from local review\n\t%#v\nexpected\n\t%#v\ngot\n\t%#v", test.description, test.localReview, &test.response, actualResponse)
+				} else {
+					failMessage = fmt.Sprintf("%s: from review\n\t%#v\nexpected\n\t%#v\ngot\n\t%#v", test.description, test.clusterReview, &test.response, actualResponse)
+				}
+				return false, nil
+			}
+
+			failMessage = ""
+			return true, nil
+		})
+
+		if err != nil {
+			t.Error(err)
 		}
+		if len(failMessage) != 0 {
+			t.Error(failMessage)
+		}
+	}
+}
 
-		failMessage = ""
-		return true, nil
-	})
+func toKubeSelfSAR(testNS string, sar *authorizationapi.LocalSubjectAccessReview) *kubeauthorizationapi.SelfSubjectAccessReview {
+	return &kubeauthorizationapi.SelfSubjectAccessReview{
+		Spec: kubeauthorizationapi.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &kubeauthorizationapi.ResourceAttributes{
+				Namespace: testNS,
+				Verb:      sar.Verb,
+				Group:     sar.Group,
+				Version:   sar.Version,
+				Resource:  sar.Resource,
+				//Subresource :sar.Subreouse,?? TODO
+				Name: sar.ResourceName,
+			},
+			//NonResourceAttributes: &kubeauthorizationapi.NonResourceAttributes{
+			//	Path: sar.Path,
+			//	Verb: sar.Verb,
+			//},
+		},
+	}
+}
 
-	if err != nil {
-		t.Error(err)
+func toKubeLocalSAR(testNS string, sar *authorizationapi.LocalSubjectAccessReview) *kubeauthorizationapi.LocalSubjectAccessReview {
+	return &kubeauthorizationapi.LocalSubjectAccessReview{
+		ObjectMeta: kapi.ObjectMeta{Namespace: testNS},
+		Spec: kubeauthorizationapi.SubjectAccessReviewSpec{
+			User:   sar.User,
+			Groups: sar.Groups.List(),
+			//Extra: scope?? TODO
+			ResourceAttributes: &kubeauthorizationapi.ResourceAttributes{
+				Namespace: testNS,
+				Verb:      sar.Verb,
+				Group:     sar.Group,
+				Version:   sar.Version,
+				Resource:  sar.Resource,
+				//Subresource :sar.Subreouse,?? TODO
+				Name: sar.ResourceName,
+			},
+			//NonResourceAttributes: &kubeauthorizationapi.NonResourceAttributes{
```

Comment:
I planned on simply removing it but I was not sure if the origin SAR stuff even handled `NonResourceAttributes`.  While both `ResourceAttributes` and `NonResourceAttributes` cannot be specified together, the origin stuff does have a `.Path` so I feel like that should be used at some point.